src/products/views/ProductList/utils.ts

```diff
@@ -0,0 +1,48 @@
+import { isInEnum } from "@saleor/misc";
+import { ProductTypeKindEnum } from "@saleor/types/globalTypes";
+import { useIntl } from "react-intl";
+
+import { productKindMessages as messages } from "./messages";
+
+interface ProductKindChoice {
+  label: string;
+  value: ProductTypeKindEnum;
+}
+
+export const getAvailableProductKinds = (): ProductKindChoice[] =>
+  Object.keys(ProductTypeKindEnum).map(kind => ({
+    label: kind,
+    value: kind as ProductTypeKindEnum
+  }));
+
+export const getProductKindOpts = (
+  availableProducts: ProductKindChoice[]
+): ProductKindChoice[] => {
+  const intl = useIntl();
```

Comment:
I would pass it as function param as this is the util function and I wouldn’t expect it may contain hooks. Potentially I could want to use util function in other not-react context, what may now cause unexpected behavior. I would be aware to only use hooks in hooks and react components.

```suggestion
export const getProductKindOpts = (
  availableProducts: ProductKindChoice[],
  intl: IntlShape
): ProductKindChoice[] => {
```

Reply:
test comment